### DISCUSSION TITLE
Using pnp and node_modules together

### DISCUSSION THREAD

**Discussion by yvuong** [yarnpkg/berry#4363] [Q&A]
We have a webapp that has been fully converted to use yarn berry's pnp. We are also trying to adopt the new SWC compiler, but it looks like their internal resolver isn't compatible with pnp, only npm or classic Yarn with the `node_modules` directory.

Is it possible to use pnp and have the `node_modules` in parallel?  As in, can we use pnp but specify certain packages we'd like to manage with the traditional `node_modules` approach?

**Comment by arcanis** [comment]
Cf this doc section: https://yarnpkg.com/getting-started/recipes#hybrid-pnp--node_modules-mono-repo

**Reply by yvuong** [reply]
thank you, I'll check it out!

**Reply by yvuong** [reply]
So if I'm interpreting the above instructions correctly, it looks like I'd have to create a whole separate directory that node_modules would live inside. Is it possible though to have the node_modules folder at the root of a package that's already using pnp?

Some extra context: we're trying to use this new compiler within an existing package.  The project is already set up with pnp.  But when we try to reference a package within the compiler's config file, which lives in the root of the package, I think the compiler's internal resolver is trying to resolve the package through the node_modules folder.  And of course, it errors out because the node_modules folder isn't there.  Here's an image in an attempt to illustrate:

![Screen Shot 2022-04-14 at 2 48 00 PM](https://user-images.githubusercontent.com/56004009/163482479-3994dfd2-b0d9-4b41-b603-7c8ad07593f4.png)
